compiler/ast.nim

```diff
@@ -502,6 +502,7 @@ type
     nfLastRead  # this node is a last read
     nfFirstWrite# this node is a first write
     nfHasComment # node has a comment
+    nfUseDefaultField
```

Comment:
There is a better way, query `sym.ast` and see if it contains a value and not nkEmpty.

Reply:
This flag is to solve field visible problem. 

```nim
  Default* = object
    poi: int = 12
    clc: Clean
    se*: range[0'i32 .. high(int32)]
```

poi is not a public attributes. For a object construct in other modules, I need to distinguish default construct and manual construct.

`Default()` should pass while `Default(poi: 15)` should raise. The ast of `Default` shouldn't be empty.

Reply:
I don't understand your remark. What does it mean "it should raise"?